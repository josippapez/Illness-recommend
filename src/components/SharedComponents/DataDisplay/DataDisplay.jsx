import React from 'react';
import PropTypes from 'prop-types';
import './DataDisplay.scss';
import classNames from 'classnames';

const DataDisplay = props => {
  return (
    <div
      style={{
        marginTop: props.removeTopSeparator ? '0px' : `${props.TopSpacing}px`,
        marginBottom: `${props.BottomSpacing}px`,
      }}
    >
      <div
        className={classNames({
          'row-display': !props.displayInColumn,
        })}
      >
        <div
          className={classNames({
            'col label': !props.displayInColumn,
            row: props.displayInColumn,
            bolded: props.headerBolded,
          })}
          hidden={!props.dataHeader}
          style={{
            fontWeight: props.headerBolded && 'bold',
            color: props.headerTextColor,
            width:
              !props.displayInColumn &&
              (props.dynamicHeaderWidth
                ? 'fit-content'
                : props.headerWidth
                  ? `${props.headerWidth}px`
                  : '150px'),
            minWidth: !props.displayInColumn && !props.headerWidth && '25%',
            marginRight: !props.displayInColumn && '16px',
            fontSize: props.headerFontSize && `${props.headerFontSize}px`,
            background: props.headerBackground,
            textTransform: props.uppercaseHeader && 'uppercase',
            alignSelf: props.centerHeaderVertically && 'center',
            textAlign: props.alignHeader ? props.alignHeader : 'left',
          }}
        >
          {props.dataHeader}
          {props.displayInColumn && !props.separateDataFromHeader && ' '}
        </div>
        {props.displayInColumn && (
          <div
            style={{
              borderTop: props.separateDataFromHeader && '1px solid #000000',
              marginTop:
                props.dataSeparatorTopSpacing &&
                `${props.dataSeparatorTopSpacing}px`,
              marginBottom:
                props.dataSeparatorBottomSpacing &&
                `${props.dataSeparatorBottomSpacing}px`,
            }}
          />
        )}
        {(props.data || props.url) && (
          <div
            className={classNames(props.dataClass, {
              col: !props.displayInColumn,
              row: props.displayInColumn && props.removeTopSeparator,
            })}
            style={{
              paddingLeft:
                !props.displayInColumn && props.dynamicHeaderWidth && '16px',
              color: props.dataDisplayColor && `${props.dataDisplayColor}`,
              width:
                props.dataFullWidth &&
                (!props.dataHeader
                  ? '100%'
                  : props.headerWidth
                    ? `calc(100% - ${props.headerWidth + 16}px)`
                    : props.floatDataRight
                      ? '100%'
                      : `calc(150 - ${16})px`),
              display: props.floatDataRight && 'flex',
              justifyContent: props.floatDataRight && 'flex-end',
            }}
          >
            {props.url ? (
              <a
                href={`//${props.url}`}
                rel="noopener noreferrer"
                target="_blank"
              >
                {props.urlName}
              </a>
            ) : (
              props.data
            )}
          </div>
        )}
      </div>
    </div>
  );
};

DataDisplay.defaultProps = {
  headerFontSize: 16,
  dataSeparatorTopSpacing: 0,
  dataSeparatorBottomSpacing: 0,
  TopSpacing: 20,
  BottomSpacing: 0,
  headerTextColor: 'black',
  alignHeader: 'left',
};

DataDisplay.propTypes = {
  dataHeader: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.node,
  ]),
  data: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.node,
  ]),
  url: PropTypes.string,
  urlName: PropTypes.string,
  headerBolded: PropTypes.bool,
  dynamicHeaderWidth: PropTypes.bool,
  displayInColumn: PropTypes.bool,
  separateDataFromHeader: PropTypes.bool,
  headerFontSize: PropTypes.number,
  dataSeparatorBottomSpacing: PropTypes.number,
  headerBackground: PropTypes.string,
  headerTextColor: PropTypes.string,
  removeTopSeparator: PropTypes.bool,
  uppercaseHeader: PropTypes.bool,
  removeTopSeparatorForData: PropTypes.bool,
  dataSeparatorTopSpacing: PropTypes.number,
  dataDisplayColor: PropTypes.string,
  dataClass: PropTypes.string,
  dataFullWidth: PropTypes.bool,
  TopSpacing: PropTypes.number,
  BottomSpacing: PropTypes.number,
  centerHeaderVertically: PropTypes.bool,
  headerWidth: PropTypes.number,
  floatDataRight: PropTypes.bool,
  alignHeader: PropTypes.string,
};

export default DataDisplay;
