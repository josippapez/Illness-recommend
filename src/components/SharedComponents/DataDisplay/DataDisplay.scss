.row-display {
  display: flex;
}
.top-separator-16 {
  margin-top: 16px;
}

.top-separator-10 {
  margin-top: 10px;
}

.top-separator-8 {
  margin-top: 8px;
}

.short-description {
  display: flex;
}

.bolded {
  font-family: "Lato, Bold";
}
