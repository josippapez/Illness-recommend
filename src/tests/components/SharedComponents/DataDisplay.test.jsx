/* eslint-disable no-undef */
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { mount } from 'enzyme';
import DataDisplay from '../../../components/SharedComponents/DataDisplay/DataDisplay';

jest.mock('../../../store/actions');

const initialState = {};

const props = {
  dataHeader: 'test',
  data: undefined,
  url: null,
  urlName: null,
  headerBolded: true,
  dataSeparatorTopSpacing: 15,
  dynamicHeaderWidth: true,
  displayInColumn: true,
  separateDataFromHeader: true,
};

const mockStore = configureStore();
const store = mockStore(initialState);

describe('Data display', () => {
  let component;

  it('renders correctly', () => {
    component = mount(
      <Provider store={store}>
        <DataDisplay {...props} />
      </Provider>
    );
    expect(component.html()).toMatchSnapshot();
  });

  it('renders correctly when url prop is passed', () => {
    const mockProps = {
      dataHeader: 'test',
      data: undefined,
      url: 'test',
      urlName: null,
      headerBolded: true,
      dynamicHeaderWidth: true,
      displayInColumn: false,
      separateDataFromHeader: true,
    };
    component = mount(
      <Provider store={store}>
        <DataDisplay {...mockProps} />
      </Provider>
    );
    expect(component.html()).toMatchSnapshot();
  });

  it('renders correctly when data prop is passed', () => {
    const mockProps = {
      dataHeader: 'test',
      data: 'test',
      removeTopSeparatorForData: true,
      displayInColumn: true,
    };
    component = mount(
      <Provider store={store}>
        <DataDisplay {...mockProps} />
      </Provider>
    );
    expect(component.html()).toMatchSnapshot();
  });
});
