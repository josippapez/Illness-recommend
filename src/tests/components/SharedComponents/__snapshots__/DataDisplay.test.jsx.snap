// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data display renders correctly 1`] = `"<div style=\\"margin-top: 20px; margin-bottom: 0px;\\"><div class=\\"\\"><div class=\\"dataDisplayHeader row bolded\\" style=\\"font-weight: bold; color: black; transition: color 0.2s ease-in-out; font-size: 16px; text-align: left;\\">test</div><div style=\\"border-top: 1px solid #000000; margin-top: 15px; margin-bottom: 0px;\\"></div></div></div>"`;

exports[`Data display renders correctly when data prop is passed 1`] = `"<div style=\\"margin-top: 20px; margin-bottom: 0px;\\"><div class=\\"\\"><div class=\\"dataDisplayHeader row\\" style=\\"color: black; transition: color 0.2s ease-in-out; font-size: 16px; text-align: left;\\">test </div><div style=\\"margin-top: 0px; margin-bottom: 0px;\\"></div><div class=\\"\\">test</div></div></div>"`;

exports[`Data display renders correctly when url prop is passed 1`] = `"<div style=\\"margin-top: 20px; margin-bottom: 0px;\\"><div class=\\"row-display\\"><div class=\\"dataDisplayHeader col label bolded\\" style=\\"font-weight: bold; color: black; transition: color 0.2s ease-in-out; min-width: 25%; margin-right: 16px; font-size: 16px; text-align: left;\\">test</div><div class=\\"col\\" style=\\"padding-left: 16px;\\"><a href=\\"//test\\" rel=\\"noopener noreferrer\\" target=\\"_blank\\"></a></div></div></div>"`;
